Source/diablo.cpp

```diff
@@ -393,8 +393,8 @@ void RightMouseDown()
 		SetSpell();
 		return;
 	}
-	if (MousePosition.y <= RIGHT_PANEL_Y || MousePosition.y >= RIGHT_PANEL_Y + SPANEL_HEIGHT
-	    || ((!sbookflag || MousePosition.x <= RIGHT_PANEL_X || MousePosition.x >= RIGHT_PANEL_X + SPANEL_WIDTH)
+	if (MousePosition.y <= rectRightPanel.position.y || MousePosition.y >= rectRightPanel.position.y + SPANEL_HEIGHT
+	    || ((!sbookflag || MousePosition.x <= rectRightPanel.position.x || MousePosition.x >= rectRightPanel.position.x + SPANEL_WIDTH)

```

Comment:
Looks like the original might have had some broken logic
```suggestion
	if ((!sbookflag || !rectRightPanel.Contains(MousePosition))
```